### PR TITLE
Added build_query_string() to URI class

### PR DESCRIPTION
**Pull request by phil-lavin**
As discussed in #fuelphp. Designed for use like this...

``` html
<a href="<?= \Uri::build_query_string(\Input::get(), ['foo'=>'bar']) ?>">Foo</a>
```

### CODE REVIEW

File: classes/uri.php [hunk 1]
```diff
@@ -228,6 +228,26 @@ public static function base($include_index = true)
...
 	}
 
+	/**
+	 * Builds a query string based on a number of merged arrays
+	 *
+	 * @param   array    Array to merge
+	 * @param   array    ...
+	 * @return  string
+	 */
+	public static function build_query_string()
```
Comment (frankdejonge) [inline classes/uri.php]: Could you add one type hinted param without a default value in the method signature? Like so: `string(array $params)`

Comment (phil-lavin) [inline classes/uri.php]: Done :)

**Comment by phil-lavin** [comment]
Yeh, except @FrenkyNet asked me to add that. Want the type hinting removed?

**Comment by phil-lavin** [comment]
At which point, it somewhat voids the point of having an argument. Want the whole argument removed?

**Comment by frankdejonge** [comment]
@WanWizard I thought it would only accept arrays as query strings are associative as well, we could remove the type hinting to allow string input too. My point was to ensure input in the first place.

**Comment by WanWizard** [comment]
No, the idea was to be able to add strings as well, if you want to drop that then the typehint is ok, but the string-to-array conversion can be removed.

Reason for wanting this is that we use GET variable a lot as toggle, so the presence of the variable is used, and not a value, like http://example.org?featureA&featureB&featureC

**Comment by phil-lavin** [comment]
@WanWizard Removed the hint

**Comment by WanWizard** [comment]
Do we have docs for this method? And preferably a unit test?

**Comment by frankdejonge** [comment]
@phil-lavin any updates on this?

**Pull request closed without merging**